src/routes/CommunitySettings/TopicsSettingsTab/TopicsSettingsTab.js

```diff
@@ -0,0 +1,239 @@
+import cx from 'classnames'
+import { find } from 'lodash/fp'
+import { boolean, arrayOf, func, number, shape, string, object } from 'prop-types'
+import React, { Component } from 'react'
+import CreateTopic from 'components/CreateTopic'
+// import { CommunityCell } from 'components/CommunitiesList/CommunitiesList'
+import Dropdown from 'components/Dropdown'
+import Icon from 'components/Icon'
+import SingleTopicSelector from 'components/TopicSelector/SingleTopicSelector'
+import ScrollListener from 'components/ScrollListener'
+import TextInput from 'components/TextInput'
+import { TOPIC_VISIBILITY } from 'store/models/Topic'
+import { inflectedTotal } from 'util/index'
+import styles from './TopicsSettingsTab.scss'
+
+const sortOptions = [
+  { id: 'name', label: 'Name' },
+  { id: 'followersTotal', label: 'Popular' }
+]
+
+const visibilityOptions = Object.keys(TOPIC_VISIBILITY).reduce((result, option) => result.concat({ value: option, label: TOPIC_VISIBILITY[option] }), [])
+
+const TOPIC_LIST_ID = 'topic-list'
+
+export default class TopicsSettingsTab extends Component {
+  state = {
+    createTopicModalVisible: false
+  }
+
+  static propTypes = {
+    community: object.isRequired,
+    defaultTopics: arrayOf(shape({
+      id: string,
+      name: string.isRequired,
+      postsTotal: number,
+      followersTotal: number,
+      isSubscribed: boolean
+    })),
+    topics: arrayOf(shape({
+      id: string,
+      name: string.isRequired,
+      postsTotal: number,
+      followersTotal: number,
+      isSubscribed: boolean
+    })),
+    network: object,
+    // routeParams: object.isRequired,
+    selectedSort: string,
+    setSort: func,
+    search: string,
+    setSearch: func,
+    setCommunityTopicVisibility: func.isRequired,
+    totalTopics: number
+  }
+
+  componentDidMount () {
+    this.props.fetchTopics()
+    this.props.fetchDefaultTopics()
+
+    // Caching totalTopics because the total returned in the queryset
+    // changes when there is a search term
+    this.setState({
+      totalTopicsCached: this.props.totalTopics
+    })
+  }
+
+  componentDidUpdate (prevProps) {
+    if (!this.state.totalTopicsCached && !prevProps.totalTopics && this.props.totalTopics) {
+      this.setState({ totalTopicsCached: this.props.totalTopics })
```

Comment:
It's a small thing but whenever I touch a setState these days I convert it to a "functional setState" such that setState({}) becomes setState(prevState => ({})) (or simply setState(() => ({})) in most use cases) . You are probably already aware of this and just updating an existing line of code and moving on, but thought it was worth mentioning. I don't know of a scenario in which not using a functional setState has bit me in the butt, but it would be a tough one to track down once it did. Better safe than sorry.

Reply:
ah yeah, thats not one that is ingrained in me yet